gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -171,46 +164,233 @@ def _get_prior_measurements_in_world_frame(
 
+        # TODO(akshay-krishnan): Use the translation covariance, transform to world frame.
+        # noise_model = gtsam.noiseModel.Gaussian.Covariance(i2Ti1_prior.covariance)
+        # TODO(akshay-krishnan): use robust noise model for priors?
+        noise_model = gtsam.noiseModel.Isotropic.Sigma(3, 1e-2)
         for (i1, i2), i2Ti1_prior in i2Ti1_priors.items():
-            # TODO(akshay-krishnan): Use the translation covariance, transform to world frame.
-            # noise_model = gtsam.noiseModel.Gaussian.Covariance(i2Ti1_prior.covariance)
-            noise_model = gtsam.noiseModel.Isotropic.Sigma(3, 1e-2)
-            w_i2ti1_priors.append(
+            w_i1ti2_prior_measurements.append(
                 BinaryMeasurementPoint3(
-                    i2,
-                    i1,
+                    C(i2),
+                    C(i1),
                     get_prior_in_world_frame(i2, i2Ti1_prior),
                     noise_model,
                 )
             )
-        return w_i2ti1_priors
+        return w_i1ti2_prior_measurements
+
+    def compute_inliers(
+        self,
+        w_i2Ui1_dict: RelativeDirectionsDict,
+        w_i2Ui1_dict_tracks: RelativeDirectionsDict,
+    ) -> Tuple[RelativeDirectionsDict, RelativeDirectionsDict, Set[int]]:
+        """Perform inlier detection for the relative direction measurements.
+
+        Args:
+            w_i2Ui1_dict: Dictionary of Unit3 relative translations between cameras.
+            w_i2Ui1_dict_tracks: Dictionary of Unit3 relative translations between cameras and landmarks.
+
+        Returns:
+            Set of indices (i1, i2) which are inliers.
+        """
+
+        # Sample directions for projection
+        combined_measurements = list(w_i2Ui1_dict.values()) + list(w_i2Ui1_dict_tracks.values())
+        projection_directions = self.__sample_projection_directions(combined_measurements)
 
-    def __get_initial_values(self, wTi_initial: List[Optional[PosePrior]]):
+        # Convert to measurements - indexes to symbols.
+        dummy_noise_model = gtsam.noiseModel.Isotropic.Sigma(3, 1e-2)  # MFAS does not use this.
+        w_i1Ui2_measurements = self._binary_measurements_from_dict(w_i2Ui1_dict, w_i2Ui1_dict_tracks, dummy_noise_model)
+
+        # Compute outlier weights using MFAS.
+        # TODO(ayush): parallelize this step.
+        outlier_weights: List[Dict[Tuple[int, int], float]] = []
+        for direction in projection_directions:
+            mfas_instance = MFAS(w_i1Ui2_measurements, direction)
+            outlier_weights.append(mfas_instance.computeOutlierWeights())
+        logger.debug("Computed outlier weights using MFAS.")
+
+        # Compute average outlier weight.
+        outlier_weights_sum: DefaultDict[Tuple[int, int], float] = defaultdict(float)
+        inliers = set()
+        for outlier_weight_dict in outlier_weights:
+            for w_i1Ui2 in w_i1Ui2_measurements:
+                i1, i2 = w_i1Ui2.key1(), w_i1Ui2.key2()
```

Comment:
Do `i1` and `i2` still mean camera indices, or can one be a landmark index? If not, can we use different variable names to disambiguate, or can we add an inline comment noting this?

Reply:
I am adding a comment that says: `'inliers' contains both camera-camera and camera-landmark inliers. We separate them here.`
Does it matter what i1 and i2 mean? within this outlier computation they are just points - could be either cameras or landmarks.

Reply:
I agree with John that i1, i2 can be a bit confusing. But I think we should be fine here.

Reply:
Maybe idx1, idx2?

Reply:
I think `w_idx2_U_idx1` might be a bit too wordy. So I'm fine with re-using `i1` and `i2` here, with the help of comments wherever we use `i1` to mean `j` (landmark)